main.go

```diff
@@ -331,3 +341,36 @@ func initKubeConfig(kcLocation string) *rest.Config {
 
 	return kubeConfig
 }
+
+func initTransport(upstreamCAFile string) (http.RoundTripper, error) {
+	if upstreamCAFile == "" {
+		return nil, nil
+	}
+
+	rootPEM, err := ioutil.ReadFile(upstreamCAFile)
+	if err != nil {
+		return nil, err
+	}
+
+	roots := x509.NewCertPool()
+	if ok := roots.AppendCertsFromPEM([]byte(rootPEM)); !ok {
+		return nil, err
+	}
+
+	transport := &http.Transport{
```

Comment:
Do we care about ALL of these settings or can we use the default transport and simply set the TLSClientConfig field?

Reply:
Yes we care, they're set for a reason. You might run into troubles, eg run out of file descriptors etc.

Reply:
@squat I misunderstood you.

We can use http.DefaultTransport direclty. I just wanted to be explicit, instead of changing types between `http.RoundTripper` and `http.Transport`.

But my understanding is that this would work:
```
	transport := http.DefaultTransport.(*http.Transport)
	transport.TLSClientConfig = &tls.Config{RootCAs: roots}

	return transport, nil
```